sklearn/metrics/pairwise.py

```diff
@@ -1265,6 +1532,28 @@ def _check_chunk_size(reduced, chunk_size):
 def _precompute_metric_params(X, Y, metric=None, **kwds):
     """Precompute data-derived metric parameters if not provided
     """
+    if metric == 'gower':
+        scale = None
```

Comment:
I'd like to see a helper pulled out from here, so that this looks like
```py
scale, min = _precompute_gower_params(X, Y, scale, min)
return {'scale': scale, 'min': min}
```
This way `gower_distances` can call `_precompute_gower_params` and the code will be more readable.